igvm/commands.py

```diff
@@ -180,6 +178,8 @@ def vm_build(vm_hostname, run_puppet=True, debug_puppet=False, postboot=None,
                 '"{}" is still running.'.format(vm.fqdn)
             )
 
+        es.enter_context(_lock_hv(vm.hypervisor))
```

Comment:
_get_best_hyprevisor should return selected HV locked. With what you proposed there is a brief moment when HV is not locked. If _get_best_hypervisor does not return locked, it must be fixed.

Reply:
The locking within this method is commented - for a reason I guess so if you are fine with adding it back in there that would be even better yes.

Reply:
I think there is no locking in _get_best_hypervisor anyway. See https://github.com/innogames/igvm/pull/71

Reply:
_get_best_hypervisor is a contextmanager so it would only release the lock after `yield` is finished.